### PR TITLE
[Sprint 2] Logic List Order

### PR DESCRIPTION
**Pull request by tranminhtri911**
https://dev.framgia.com/redmine/issues/112994
Logic List Order
Effect : None

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    @Override
+    public int getItemViewType(int position) {
+        if (mShoppingCartList.get(position) instanceof ShoppingCart) {
+            return HEADER;
+        } else {
+            if (mShoppingCartList.get(position) instanceof ShoppingCart) {
+                return CONTENT;
+            }
+        }
+        return -1;
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java]: Please change to
```java
         if (mShoppingCartList.get(position).isForHeader() {
             return HEADER;
         } 
         return CONTENT;
         }
```

File: app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+public class ShoppingCartAdapter extends BaseExpandableListAdapter {
+    private static OrderItemListener mOrderListener;
+    private Context mContext;
+    private List<Cart> mCartList;
+
+    protected ShoppingCartAdapter(@NonNull Context context, List<Cart> cart) {
+        mContext = context;
+        mCartList = new ArrayList<>();
+        if (mCartList == null) {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java]: when mCartList ==null?

Comment (DoanVanToan) [inline app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java]: pls check null of cart

File: app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            return;
+        }
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java]: Remove this condition, It unnecessary, ``mCartList`` initialized above

File: app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    @Override
+    public boolean hasStableIds() {
+        return false;
+    }
+
+    @Override
+    public View getGroupView(int groupPosition, boolean isExpanded, View convertView,
+            ViewGroup parent) {
+        if (convertView == null) {
+            LayoutInflater layoutInflater = (LayoutInflater) mContext.
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java]: change layoutInflater to instance variable

File: app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                    getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+            convertView = layoutInflater.inflate(R.layout.item_header_cart, null);
+        }
+        ItemHeaderCartBinding binding = DataBindingUtil.bind(convertView);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java]: change to 
```java
ItemHeaderCartBinding binding = DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()),
                        R.layout.item_header_cart, parent, false);
```

Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartAdapter.java]: you can refer ``ProductAdapter``

File: app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartPresenter.java [hunk 6]
```diff
@@ -1,24 +1,143 @@
...
+    @Override
+    public void downQuantity(CartItem cartItem) {
+        Domain domain = mDomainRepository.getCurrentDomain();
+        if (cartItem == null || domain == null) {
+            return;
+        }
+        Subscription subscriptions =
+                mProductRepository.downQuantity(cartItem.getProductId(), domain.getId())
+                        .subscribe(new Action1<Void>() {
+
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartPresenter.java]: remove empty line

File: app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartPresenter.java [hunk 7]
```diff
@@ -1,24 +1,143 @@
...
+    @Override
+    public void deleteProduct(CartItem cartItem) {
+        Domain domain = mDomainRepository.getCurrentDomain();
+        if (cartItem == null || domain == null) {
+            return;
+        }
+        Subscription subscriptions =
+                mProductRepository.deleteCart(cartItem.getProductId(), domain.getId())
+                        .subscribe(new Action1<Void>() {
+
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/cart/ShoppingCartPresenter.java]: remove empty line

**Pull request merged**